jdaviz/core/template_mixin.py

```diff
@@ -281,24 +281,42 @@ def hub(self):
     def viewer_dicts(self):
         def _dict_from_viewer(viewer, viewer_item):
             d = {'viewer': viewer,
-                 'id': viewer_item['id'],
-                 'icon': self.app.state.viewer_icons.get(viewer_item['id'])}
+                 'id': viewer_item.get('id'),
+                 'icon': self.app.state.viewer_icons.get(viewer_item.get('id'))}
             if viewer_item.get('reference') is not None:
-                d['reference'] = viewer_item['reference']
-                d['label'] = viewer_item['reference']
+                d['reference'] = viewer_item.get('reference')
+                d['label'] = viewer_item.get('reference')
             else:
                 d['reference'] = None
-                d['label'] = viewer_item['id']
+                d['label'] = viewer_item.get('id')
 
             return d
 
         return [_dict_from_viewer(viewer, self.app._viewer_item_by_id(vid))
                 for vid, viewer in self.app._viewer_store.items()
                 if viewer.__class__.__name__ != 'MosvizTableViewer']
 
-    @cached_property
+    @property
```

Comment:
does the caching logic fail because the viewer name might change during runtime? 😬